src/components/PinningAlerts.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from "react";
+import Snackbar from "@material-ui/core/Snackbar";
+import MuiAlert from "@material-ui/lab/Alert";
+import LinearProgress from "@material-ui/core/LinearProgress";
+
+//PinningAlerts component shows alerts when pinning a post from the blog view
+export default function PinningAlerts(props) {
+  /*
+        props -> pinStatus, setPinStatus
+        showPinning -> state for handling pinning notif
+        showUnpinning -> state for handling unpinning notif
+        showSuccess -> state for handling success notif
+        showError -> state for handling error notif
+         */
+  const { pinStatus, setPinStatus } = props;
+  const [showPinning, setShowPinning] = useState(false);
+  const [showUnpinning, setShowUnpinning] = useState(false);
+  const [showSuccess, setShowSuccess] = useState(false);
+  const [showError, setShowError] = useState(false);
+
+  //effect that updates states when pinstatus changes
+  useEffect(() => {
+    setShowPinning(false);
+    setShowUnpinning(false);
+    setShowSuccess(false);
+    setShowError(false);
+    if (pinStatus === "pinning") {
```

Comment:
Follow up: I think it is good practice to use a switch statement here so that you can have a `default` case to catch potential bugs where pinStatus is not one of the options set here. It also helps catch bugs like typos elsewhere in the code.